### ISSUE TITLE
[Deque Analysis] Thumbnail images must have alt text

### ISSUE THREAD

**Issue by tdonohue** [DSpace/dspace-angular#1169]
> Critical: This issue results in blocked content for people with disabilities, and will definitely prevent them from accessing fundamental features or content. This type of issue puts your organization at risk. Prioritize fixing as soon as possible, within the week if possible. Remediation should be a top priority. 

## Deque Analysis Summary

A total of 2 Critical issues exist in the Item/Entity pages, both having to do with alt text on thumbnails:

1. (IDs 469681, 469997) "No Thumbnail" image needs "alt" text, e.g. https://demo7.dspace.org/entities/publication/4e203020-8fd6-406a-addc-f4f08365b662  (In this case alt text could just say "No thumbnail available")
     * This includes placeholder images on entity pages, e.g. https://demo7.dspace.org/entities/project/b1bc3a49-49b1-417a-ac90-8d5c7ba5e0ac  (In this case alt text could just say "Project placeholder image" or similar)
     * **UPDATE:** In a separate Deque Issues (ID: 469763, 469998), they noted it's an issue that "Image of text is used instead of real text".  They stated that these inline "No thumbnail" images might be better replaced by simple HTML text that is styled with CSS to look like an image... as that'd negate the need for alt text or even downloading an image here. So, this either needs alt text or needs to be regular HTML....either solution is fine, and we could start with one and improve to the other at a later time.
2. (ID 470938) Existing thumbnail images also need "alt" text, e.g. https://demo7.dspace.org/entities/person/5a3f7c7a-d3df-419c-b8a2-f00ede62c60a  (In this case alt text could just say "Thumbnail image of [entity type or Item]" or similar)

The full list of these issues (which includes links for more info) can be found at this link (requires login): 
https://axeauditor.dequecloud.com/test-run/0856438a-a19a-11eb-bc31-b7d5be387c86/issues?activeTab=dt-issue&page=0&pageSize=100&sortField=ordinal&sortDir=asc&row=9&filter%5Bseverity%5D=4&filter%5Bpage_number%5D=6&filter%5Bpage_number%5D=7

## More Information / Tools

* Consider using [Deque's free Chrome plugin](https://chrome.google.com/webstore/detail/axe-devtools-web-accessib/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US) to check your work

**Comment by tdonohue** [comment]
@artlowel : Assigning this to you/your team for analysis.

**Comment by artlowel** [comment]
- Add inputs to ds-thumbnail to specify an i18n key to use as alt text for when the thumbnail exists, and when it's missing. 
- Default to "No Thumbnail available" and "Thumbnail Image" when that input isn't used
- Use those inputs on entity pages to provide different values (e.g. "Project placeholder image")

This will take an estimated 3 hours

**Comment by tdonohue** [comment]
Sounds good...obviously, alt text is not needed when the thumbnail doesn't exist (since in #1162 you noted we'll switch to HTML approach).  However, I agree with your overall note here...we should make alt text (when image exists) a key in i18n.  When image doesn't exist, the HTML should also use an i18n key.  Will add estimate & keep assigned to you/your team
